### ISSUE TITLE
Function for submitting the form programatically?

### ISSUE THREAD

**Issue by winston0410** [pablo-abc/felte#58]
Hi, I want to submit the form in an event listener, but I cannot find any function for that in the api doc.

https://api.felte.dev/

I have then tried out the traditional `form.submit()` (and reference the `<form>` element with `bind:this`) in Javascript, but seems like Felte is not catching the submit there.

```typescript
const handleKeydown = (e: KeyboardEvent) => {
    if(e.code === "Enter"){
        formElem.submit()
    }
}
```

Did I missed something in the documentation? Or is this function not there yet? I would love to help with the doc or the code.

**Comment by pablo-abc** [comment]
Hello! https://api.felte.dev right now is more aimed at people trying to make sense of all the helper libraries in order to expand them rather than for people just trying to _use_ Felte.

For your specific use case, you may use the [`createSubmitHandler`](https://felte.dev/docs/svelte/helper-functions#createsubmithandler) helper function!

Regarding `form.submit()`, traditionally calling that _does not_ trigger a `submit` event, neither does it call `onsubmit` handlers, that's why Felte's not able to catch it. Overriding the method so it uses Felte could be possible, but since that use case is covered by `createSubmitHandler` I didn't think it'd be worth it.

**Comment by pablo-abc** [comment]
As an addendum to this, MDN docs seem to indicate the proper way to do this (by calling a method from the `form` element) is by using [HTMLFormElement.requestSubmit](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit). I see it actually does not work on Safari. So I've just published a patch on `felte@0.8.3` that just calls our submit handler if the browser does not support it or no polyfill has been added.
So calling `form.requestSubmit()` should work for your use-case, _and_ starting on 0.8.3 it also works in Safari.
